core-shaded/pom.xml

```diff
@@ -0,0 +1,103 @@
+      <plugin>
+        <artifactId>maven-shade-plugin</artifactId>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <artifactSet>
+                <includes>
+                  <include>com.datastax.oss:java-driver-core</include>
+                  <include>io.netty:*</include>
+                </includes>
+              </artifactSet>
+              <relocations>
+                <relocation>
+                  <pattern>io.netty</pattern>
+                  <shadedPattern>com.datastax.oss.driver.shaded.netty</shadedPattern>
+                </relocation>
+              </relocations>
+              <transformers>
+                <transformer implementation="org.apache.maven.plugins.shade.resource.DontIncludeResourceTransformer">
+                  <resources>
+                    <resource>META-INF/MANIFEST.MF</resource>
+                    <resource>META-INF/maven/com.datastax.oss/java-driver-core/pom.xml</resource>
+                    <resource>META-INF/maven/com.datastax.oss/java-driver-core/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-handler/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-handler/pom.xml</resource>
+                    <resource>META-INF/maven/io.netty/netty-buffer/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-buffer/pom.xml</resource>
+                    <resource>META-INF/maven/io.netty/netty-common/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-common/pom.xml</resource>
+                    <resource>META-INF/maven/io.netty/netty-transport/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-transport/pom.xml</resource>
+                    <resource>META-INF/maven/io.netty/netty-resolver/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-resolver/pom.xml</resource>
+                    <resource>META-INF/maven/io.netty/netty-codec/pom.properties</resource>
+                    <resource>META-INF/maven/io.netty/netty-codec/pom.xml</resource>
+                    <!-- netty's shading of jctools does not remove its pom files -->
+                    <resource>META-INF/maven/org.jctools/jctools-core/pom.properties</resource>
+                    <resource>META-INF/maven/org.jctools/jctools-core/pom.xml</resource>
+                  </resources>
+                </transformer>
+                <!-- Pick up the alternate manifest that was generated by the core module -->
+                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
+                  <resource>META-INF/MANIFEST.MF</resource>
+                  <file>${project.parent.basedir}/core/target/META-INF-shaded/MANIFEST.MF</file>
```

Comment:
This might be a bit suspect/non-ideal, what do you think?  We rely on the java-driver-core module to generate the shaded manifest for us, and then pull it in here.  I found this a better alternative than explicitly defining the `Export-Package` and `Import-Package` sections which is quite complex.

Reply:
As discussed privately, the relative reference bothers me a bit out of principle: in theory, you should be able to check out a module as an independent Maven project and build it (provided that the parent POM and all upstream modules are in your local repo).

But, all things considered, there's little chance of that happening, so that's the most pragmatic approach here.

Reply:
The other possible viable alternative is to add the shaded MANIFEST to the core jar somwhere, and then pull that MANIFEST from the jar when we shade and put it into `META-INF/MANIFEST.MF`.  That would work around the problem of relative referencing, but will add a unrelated file to the core jar.  Is that worth doing or would you rather not add that extra file to the core jar?

Reply:
What does it take to pull the file from the other JAR, just a different transformer? If it's a similar amount of lines in the POM, I don't mind adding an unrelated file to the core jar. If it's more than that, we can live with the relative reference.

Reply:
It's just as easy, and more straightforward (as doesn't rely on location of generating file in other module).  I can make that change.

Reply:
Done: 3d32d0017 . Wasn't as easy as I anticipated but no longer relies on relative pathing.